### ISSUE TITLE
Sunsetting Node 14 and 16 support

### ISSUE THREAD

**Issue by wopian** [wopian/kitsu#847]
Guaranteed Node 14 and 16 support with be removed once the TypeScript rewrite has been completed in #818 as the following Node versions become EOL in 2023:

- Node 14: [30th April 2023](https://github.com/nodejs/release)
- Node 16: [11th September 2023](https://github.com/nodejs/release)

Applications that are stuck on using Node 14 will likely continue to function as normal as I don't think there has been any major language features added in Node 16+/modern browsers that are in use by our code that is potentially being polyfilled for compatibility purposes by Babel.

**Comment by wopian** [comment]
Due to the timeframe the TypeScript rewrite has taken, Node 16 will also be sunset with it.

https://github.com/wopian/kitsu/pull/818

@pedep feel free to assist migrating functionality over if you have time. My availability has been slim between work, Zeepkist and my photography work over the past several months 😿

**Comment by pedep** [comment]
@wopian i have begun refactoring parts of kitsu-core in https://github.com/wopian/kitsu/pull/930 . Please let me know if you see anything you would like changed.

**Comment by pedep** [comment]
@wopian any thoughts on removing axios as a direct dependency of `kitsu`, in favor of using native fetch?
Since we are sunsetting node < 18, we can use fetch native both client and servers-side.

I would suggest moving an axios implementation to a separate package `kitsu-axios` for everyone with axios specifics in their codebase, such as middleware.

```typescript
import { Kitsu } from 'kitsu'
import { AxiosAdapter } from 'kitsu-axios'

const api = new Kitsu({adapter: new AxiosAdapter()})

// or

Kitsu.default_adapter = AxiosAdapter;
const api = new Kitsu()
```

Another option is to add the `kitsu-axios` package as a dependency of `kitsu` and set it as the default. Then we can swap the defaults and remove the direct dependency in a later version. This way the current axios implementation sticks around until everyone has had a chance to migrate to fetch, or add a direct dependency on `kitsu-axios` to continue using axios

```typescript
import { Kitsu, FetchAdapter } from 'kitsu'

const api = new Kitsu({adapter: new FetchAdapter()})
```

**Comment by wopian** [comment]
@pedep yes, `^11` would be a good opportunity to finally get rid of Axios.

The `kitsu` package is relatively basic so I don't think we'd need to worry about implementing an adapter - it's only a few lines of code outside of all the JSDoc comments. 

`kitsu` could be soley native fetch from `^11` onwards and `kitsu-axios` would be an identical implementation with Axios `^1.x` instead. So your upgrade paths from `kitsu@9` and `kitsu@10` (both Axios `^0.x`) are either native Fetch or Axios `^1`

**Comment by wopian** [comment]
Eslint v9 dev dependency drops support for Node 16, so this package will be pinned to v8 until Kitsu v11

**Comment by wopian** [comment]
`kitsu@11` has dropped Node 14 and 16 support. For now it remains an axios wrapper (now v1 instead of v0). Refactoring from axios to native fetch will be moved to `kitsu@12` or higher.
